### PR TITLE
refactor find first empty

### PR DESCRIPTION
**Pull request by fahro**
I think this refacotring is much better for your code. bla bla bla...

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,3 @@
 sudoku
-__pycache__
\ No newline at end of file
+__pycache__
+.vscode
```
Comment (brodude456) [inline .gitignore:3]: why did you add that m8

Comment (fahro) [reply .gitignore:3]: Because I am using vscode tool and I dont want to add autogenerated files...

Comment (brodude456) [reply .gitignore:3]: ok

File: main.py [hunk 2]
```diff
@@ -17,88 +17,97 @@
...
+            bo[previus_not_filled_place[0]][previus_not_filled_place[1]] = 0
 
     return None
 
+
 def valid(bo, num, pos):
 
-    if num!=0:
-        for colind,col in enumerate(bo[pos[0]]):
-            if col==num and colind!=pos[1]:
```
Comment (brodude456) [inline main.py:76]: this was better

Comment (fahro) [reply main.py:76]: good

Comment (brodude456) [reply main.py:76]: but lets keep the new one
